test/tests/functional/pbs_reservations.py

```diff
@@ -120,6 +120,35 @@ def test_degraded_standing_reservations(self):
         self.server.expect(RESV, a, id=rid, interval=1, offset=offset,
                            attrop=PTL_AND)
 
+    def test_duration_end_resv(self):
+        """
+        To test if reservations can be made by using
+        Duration and EndTime, making the server calculate
+        the StartTime.
+        """
+        # create reservation to end in 5 seconds and lasts 1.
+        a = {'Resource_List.select': '1:ncpus=1',
+             'reserve_end': int(time.time()) + 5,
+             'reserve_duration': 1}
+        R = Reservation(TEST_USER, attrs=a)
+        R.unset_attributes(['reserve_start'])
+        rid = self.server.submit(R)
+
+        a = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        self.server.expect(RESV, a, id=rid)
+
+        # create reservation to end in 5 seconds and lasts 10.
+        a = {'Resource_List.select': '1:ncpus=1',
+             'reserve_end': int(time.time()) + 5,
+             'reserve_duration': 10}
+        R = Reservation(TEST_USER, attrs=a)
+        R.unset_attributes(['reserve_start'])
+        try:
+            rid = self.server.submit(R)
+        except PbsSubmitError, e:
+            self.assertTrue('pbs_rsub: Bad time specification(s)' in e.msg[0],
+                            'Reservation Submission failed in an unexpected way')
```

Comment:
So these negative tests require you to test both sides.  You are just testing the correct side.  You submit a reservation with bad input and got the expected output.  What happens if the reservation was successfully submitted?  Your test would still pass.  You need to set rid = None before the try and check that rid is still None after try/except.